src/R/Components/Impl/InteractiveWorkflow/Implementation/RInteractiveEvaluator.cs

```diff
@@ -99,7 +99,13 @@ public sealed class RInteractiveEvaluator : IInteractiveEvaluator {
 
         public async Task<ExecutionResult> ResetAsync(bool initialize = true) {
             try {
+                bool isResetting = true;
                 if (Session.IsHostRunning) {
+                    if(MessageButtons.Yes == await _coreShell.ShowMessageAsync(Resources.Warning_SaveOnReset, MessageButtons.YesNo)) {
+                        await Session.ExecuteAsync("rtvs:::save_state()", REvaluationKind.NoResult);
```

Comment:
You don't need `NoResult`, it's implied by the use of `ExecuteAsync` rather than `EvaluateAsync`.

And yes, this should catch exceptions and log to REPL. Something along the lines of "failed to save state, resetting anyway".